### PR TITLE
Fix remaining Java Serialization issues with JDK21

### PR DESCRIPTION
**Pull request by KrnSaurabh**
Fix remaining Java Serialization issues with JDK21

### CODE REVIEW

File: generator/Generator.sc [hunk 1]
```diff
@@ -1131,13 +1131,18 @@ def generateMainClasses(): Unit = {
                   case 2 => BiFunctionType
                   case _ => s"Function$i"
                 }
+                val accessModifier = i match {
+                  case 1 => "transient final"
+                  case 2 => "transient final"
+                  case _ => "final"
```
Comment (pivovarit) [inline generator/Generator.sc:1137]: Can't it be `transient final` everywhere?

Comment (KrnSaurabh) [reply generator/Generator.sc:1137]: Yes we can keep `transient final` for all `Case`, Although `Case3` onwards uses `io.vavr.Function` which is already serializable, hence there is no serializable warning for `Case3` onwards.

Let me know if I should keep `transient final` for all `Case`.

Comment (pivovarit) [reply generator/Generator.sc:1137]: Ah, thanks for the clarification - this is ok

**Review by pivovarit — APPROVED** [review]

**Comment by KrnSaurabh** [comment]
@pivovarit PR to backport in version/0.x branch --> https://github.com/vavr-io/vavr/pull/2880

**Comment by pivovarit** [comment]
@KrnSaurabh looking now, trying to understand what fails when building with JDK 23

**Pull request merged**